ipc/provider/src/manager/evm/manager.rs

```diff
@@ -136,22 +130,28 @@ impl TopDownFinalityQuery for EthSubnetManager {
             .topic1(topic1);
 
         let mut messages = vec![];
+        let mut hash = None;
         for (event, meta) in query_with_meta(ev, gateway_contract.client()).await? {
-            let msg = CrossMsg::try_from(event.message)?;
-            log::debug!("received message: {:?} and meta: {:?}", msg, meta);
-
-            if meta.block_hash.0 != block_hash {
-                return Err(anyhow!(
-                    "block hash not equal, input: {}, received: {}",
-                    hex::encode(block_hash),
-                    hex::encode(meta.block_hash.0)
-                ));
+            if let Some(h) = hash {
```

Comment:
Is this check still needed?

Reply:
yeah, we have to ensure the events have the same block hashes.